proxy/http/HttpSM.cc

```diff
@@ -3501,15 +3511,8 @@ HttpSM::tunnel_handler_post_ua(int event, HttpTunnelProducer *p)
       SMDebug("http_tunnel", "send 408 response to client to vc %p, tunnel vc %p", ua_txn->get_netvc(), p->vc);
 
       tunnel.chain_abort_all(p);
-      server_session = nullptr;
```

Comment:
IIUC, the crash will be fixed by setting `server_entry->in_tunnel = false` here.
I'm a bit afraid that this PR is too generic for fixing a crash.